### PR TITLE
[871] Remember ERC20 approval

### LINKED ISSUE #871: Remember if ERC20s have been approved
per [this Discord thread](https://discord.com/channels/562828676480237578/1053196978793750559/1054364500956041236)

**User Story:**
As a
I want to
So that

**Acceptance Criteria**
GIVEN
WHEN
THEN

**Product & Design Links:**

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by DanieleSalatti**
##### Description

Skips the approval step if not necessary.

##### Refers/Fixes

Closes #871 

##### Testing

Tested locally donating to current production round

### CODE REVIEW

File: packages/grant-explorer/package.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^13.0.1",
     "@testing-library/user-event": "^14.1.1",
-    "allo-indexer-client": "github:gitcoinco/allo-indexer-client#0487c02c9bb89307b12ffe75f5bfb578278994a1",
+    "allo-indexer-client": "github:gitcoinco/allo-indexer-client#48490969985125e89bf1d65725ef9e510c97256a",
```
Comment (thelostone-mc) [inline packages/grant-explorer/package.json:68]: should revert this change ? :P

**Review by willsputra — APPROVED** [review]

**Review by michellema1208 — APPROVED** [review]
I don't have ETH or DAI in my test wallet atm so I didn't test, but Will tested on behalf of PM/UX and confirmed it's working!

**Review by vacekj — APPROVED** [review]
Are the formatting changes coming from prettier? Or some default IDE formatter?

File: packages/grant-explorer/package.json (hunk 1 above)
Comment (thelostone-mc) [inline packages/grant-explorer/package.json:68]:
```suggestion
    "allo-indexer-client": "github:gitcoinco/allo-indexer-client#0487c02c9bb89307b12ffe75f5bfb578278994a1",
```
